app/components/animated_number/index.test.tsx

```diff
@@ -0,0 +1,176 @@
+//     // }
+
+//     const MockText = jest.fn((props) => (<Text
+//         {...props}
+//         onLayout={() => ({nativeEvent: {layout: {height: 10}}})}
+//     />));
+
+//     return MockText;
+// });
+
+// no longer used for testing, but left for comparison
+describe.skip('AnimatedNumber, using toMatchSnapshot', () => {
+    it('should render correctly', () => {
+        const {toJSON} = render(<AnimatedNumber animateToNumber={123}/>);
+        expect(toJSON()).toMatchSnapshot();
+    });
+
+    it('should rerender the correct number', () => {
+        const {toJSON, getByTestId, rerender} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        expect(toJSON()).toMatchSnapshot();
+
+        const text = getByTestId('no-animation-number');
+
+        fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: 10}}});
+
+        rerender(<AnimatedNumber animateToNumber={124}/>);
+
+        rerender(<AnimatedNumber animateToNumber={125}/>);
+
+        expect(toJSON()).toMatchSnapshot();
+    });
+});
+
+const NUMBER_HEIGHT = 10;
+
+describe('AnimatedNumber', () => {
+    // running on jest, since Animated is a native module, Animated.timing.start needs to be mocked in order to update to the final Animated.Value.
+    // Ex: 1 => 2, the Animated.Value should be -20 (from -10) after the animation is done
+    jest.spyOn(Animated, 'timing').mockImplementation((a, b) => ({
+
+        // @ts-expect-error mock implementation for testing
+        start: jest.fn().mockImplementation(() => a.setValue(b.toValue)),
+    }) as unknown as Animated.CompositeAnimation);
+
+    it('should render the non-animated number', () => {
+        const {getByTestId} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        const text = getByTestId('no-animation-number');
+        expect(text.children[0]).toEqual('123');
+    });
+
+    it('should removed the non-animation number after getting the correct height', () => {
+        const {getByTestId, queryByTestId} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        const text = getByTestId('no-animation-number');
+
+        fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+        const removedText = queryByTestId('no-animation-number');
+
+        expect(removedText).toBeNull();
+    });
+
+    it('should switch to the animated number view', async () => {
+        const {getByTestId} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        const text = getByTestId('no-animation-number');
+
+        fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+        const animatedView = getByTestId('animation-number-main');
+        expect(animatedView).toBeTruthy();
+    });
+
+    describe.each([1, 23, 579, -123, 6789, 23456])('should show the correct number of animated views based on the digits', (animateToNumber: number) => {
+        it(`should display ${animateToNumber.toString().length} view(s) for ${animateToNumber}`, async () => {
```

Comment:
We might want to replace any occurrences of `X.toString().length` with a helper method like `countDigits(X)` since it took me a minute to remember what that did. Giving it a name would help make it clearer what we're trying to do with it.

On my first read of this, I thought this meant that it made `animateToNumber` views, and I was going to comment on how that would cause problems when there 😅